src/microbe_stage/MicrobeStage.cs

```diff
@@ -117,6 +119,14 @@ public class MicrobeStage : NodeWithInput, IReturnableGameState, IGodotEarlyNode
     [JsonIgnore]
     public PlayerHoverInfo HoverInfo { get; private set; } = null!;
 
+    [JsonProperty]
+    public WorldGenerationSettings WorldSettings
```

Comment:
Shouldn't this be json ignore and just have a getter for getting the object from the gameworld? Similarly to how the gameworld is just a helper property to read it from the current game.

Reply:
The issue then is adding the world settings to the MicrobeStage object, which I did in NewGameSettings.OnConfirmPressed(). As it stands, trying to set the WorldSettings variable from GameWorld in the same place throws an error since CurrentGame is null at that point.

Reply:
CurrentGame is not initialised at the time we exit the game settings screen. It is only added to MicrobeStage in SetupStage(), which is called after we switch to the scene. So the key problem is that we can't just pass the world settings to CurrentGame at this point.

I tried passing the settings to the new game _after_ switching the scene, but then many things (starting patch, cloud densities, etc.) initialised without the correct world settings.

I agree that GameWorld should be the source of truth for the world settings, but I can't figure out a way of initialising everything correctly without passing a WGS object to MicrobeStage itself. So I've changed this variable to InitialWorldSettings, used only for this initial setup. Everything else should use the WorldSettings object in GameWorld.

Reply:
I couldn't help myself so I went ahead and programmed and committed this kind of change. Also added 2 more words to the dictionary to suppress some warnings.